lib/managers/helpers/TranscodingHelper.js

```diff
@@ -38,50 +39,65 @@ class TranscodingHelper {
 		});
 	}
 
-	static transcodeAndSaveToDisk(flavorId, jsonInfo, durationInSeconds, apiConnector, partnerId, inPath, outDir)
+	static transcodeAndSaveToDisk(flavorId, jsonInfo, durationInSeconds, apiConnector, partnerId, inPath, outDir, isAd = false)
 	{
-
 		return new Promise( function(resolve, reject)
 		{
 			function gotCommandResponse(response)
 			{
-				const namespace = continuationLocalStorage.getNamespace('play-server');
-				fs.access(response.path, fs.constants.F_OK,
-					namespace.bind(function (err)
+				KalturaTempFileHandler.getFileStatus(response.path).then(function (status)
+				{
+					switch (status)
 					{
-						if (err === null)
-						{
+						case FILE_STATUS.READY:
 							KalturaLogger.log(`File existed on local disk ${response.path}`);
 							resolve(new TranscodedFileInfo(response.path, durationInSeconds));
-						}
-						else
-						{
-							gTranscodingEngine.transcodeFile(response.command, flavorId, response.path).then(
-								function (data)
-								{
-									KalturaLogger.log(`Managed to transcode and save file ${response.path}`);
-									resolve(new TranscodedFileInfo(response.path, durationInSeconds));
-								}, function (err)
+							break;
+						case FILE_STATUS.PROCESSING:
+							reject(`${ERROR_FILE_IS_TRANSCODING} ${response.path}`);
+							break;
+						case FILE_STATUS.DOESNT_EXIST:
+							KalturaUtils.createFilePath(response.path,
+								function(){
+									try
+									{
+										fs.writeFileSync(`${response.path}.tmp`, '', { flag: 'wx' });
```

Comment:
I assume that you have the .tmp somewhere else - call both of them from the constants